core/src/test/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1238,7 +1238,7 @@ public void restartRandomDataNode(RestartCallback callback) throws Exception {
     /**
      * Restarts a random node in the cluster and calls the callback during restart.
      */
-    private void restartRandomNode(Predicate<NodeAndClient> predicate, RestartCallback callback) throws Exception {
+    public void restartRandomNode(Predicate<NodeAndClient> predicate, RestartCallback callback) throws Exception {
```

Comment:
I now added a new method in InternalTestCluster that restarts a dedicated node with a callback and made all other things private again. Let me know if this works for you.